providers/gce/gce_loadbalancer_internal.go

```diff
@@ -254,6 +256,37 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 	return status, nil
 }
 
+func filterOutNodesFromNonDefaultNetworks(nodes []*v1.Node, defaultSubnetURL string) []*v1.Node {
```

Comment:
Can we change the name, it's very long and it's not specific to defaultSubnet.  I recommend changing this func to this:

```
func removeNodesWithSubnetName(nodes []*v1.Node, subnetName string) []*v1.Node.
```

This should have its own unit test (TestFilterOutNodesFromN...)

Parse the subnet URL outside this func.

Reply:
This function is actually very specific to the default subnet. It filters using the provided subnet name but also on some conditions that are very specific only to the default subnet.
1. The PodCIDR fallback check - this is needed for backwards compatibility. If for some reason the labels are not added by the node controller we must be able to add nodes. PodCIDR presence check was advised to us for this since the same controller will add it. (PodCIDR is there -> node controller processed the node -> either the label is set or it's the default network)
2. The empty label case - apparently this is intended for nodes in the default network case, I think when the cluster is not using multi subnet yet.